### PR TITLE
[tcl] Add new port

### PR DESCRIPTION
**Pull request by greenbrettmichael**
Support tcl core in vcpkg. related https://github.com/microsoft/vcpkg/issues/7647

### CODE REVIEW

File: ports/tcl/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        file(COPY ${CURRENT_PACKAGES_DIR}/bin/tclsh90s.exe DESTINATION ${CURRENT_PACKAGES_DIR}/tools/tcl/bin)
+        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin)
+    endif()
+    vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/tcl/bin)
+    file(REMOVE ${CURRENT_PACKAGES_DIR}/bin/tclsh90.exe)
+    message(STATUS "Installing ${TARGET_TRIPLET}-release done")
+
+    file(INSTALL ${SOURCE_PATH}/license.terms DESTINATION ${CURRENT_PACKAGES_DIR}/share/tcl RENAME copyright)
+else()
+    message(ERROR "Unsupported Operating System ${VCPKG_CMAKE_SYSTEM_NAME}")
```
Comment (cbezault) [inline ports/tcl/portfile.cmake]: While I appreciate you explicitly failing on platforms you didn't develop for, I know for a fact that TCL is cross-platform. Is the effort in getting it to work with linux and osx too high?

Comment (greenbrettmichael) [reply ports/tcl/portfile.cmake]: @cbezault  you are absolutely right! tcl has a custom config for linux https://github.com/tcltk/tcl/tree/master/unix  and osx https://github.com/tcltk/tcl/tree/master/macosx.

unfortunately I didn't have much luck implementing on my Ubuntu VM, I believe the configuration needs a treatment similar to ffmpeg https://github.com/microsoft/vcpkg/tree/master/ports/ffmpeg , but was unable to apply this.

as far as osx goes I do not have a VM for that OS and so I cannot push a OS specific configuration with confidence. I can run it against the azure pipelines but that doesn't help me know if the package is usable after configuration.

Comment (greenbrettmichael) [reply ports/tcl/portfile.cmake]: @cbezault I would also like to add that I do not know if ARM configuration is possible on Windows, see this line https://github.com/tcltk/tcl/blob/master/win/makefile.vc#L96

I know that ARM is definitely possible on Linux as I have seen builds for that configuration online

Comment (cbezault) [reply ports/tcl/portfile.cmake]: Yeah last I remember I couldn't run Tcl on ARM. We'd have people emulate x86 on their arm devices if they really needed to run on them.

**Comment by cbezault** [comment]
@grdowns, I'm inclined to merge this PR even if it's not perfect.

**Review by cbezault — APPROVED** [review]

**Comment by cbezault** [comment]
@greenbrettmichael ping.

**Comment by greenbrettmichael** [comment]
@cbezault  I signed the CLA, sorry I didn't see your edit

**Comment by Voskrese** [comment]
>  Version: 8.6.5

>  # the working 9.0 alpha release has fixes to their nmake script that are needed, 8.6.9 has issues with configuration

https://github.com/tcltk/tcl/tree/fb28af5fa6c4ffcd2d176c5617e5640acbfb8114
**This is the Tcl 9.0a0 source distribution.**

**Comment by cbezault** [comment]
Good point the version string should be changed

**Pull request merged**